### ISSUE TITLE
Internet radio export Unexpected error

### ISSUE THREAD

**Issue by poltrian** [owncloud/music#822]
I Try to Export my list. Goto Export in a Files and it shows Unexpected error

**Comment by paulijar** [comment]
Thanks for the report. When this happens, can you find any more information from nextcloud.log or owncloud.log (depending on which cloud you use)?

**Comment by poltrian** [comment]
Hope that helps.

Nextcloud 20.0.7 PHP 7.4 Ubuntu 20.04

  | Exception: Argument  3 passed to OCA\Music\Controller\RadioApiController::exportAllToFile()  must be of the type string, int given, called in  /var/www/html/nextcloud/lib/private/AppFramework/Http/Dispatcher.php on  line 169/var/www/html/nextcloud/lib/private/AppFramework/App.php - line 152: OC\AppFramework\Http\Dispatcher->dispatch()/var/www/html/nextcloud/lib/private/Route/Router.php - line 309: OC\AppFramework\App::main()/var/www/html/nextcloud/lib/base.php - line 1008: OC\Route\Router->match()/var/www/html/nextcloud/index.php - line 37: OC::handleRequest()Verursacht durchTypeError: Argument  3 passed to OCA\Music\Controller\RadioApiController::exportAllToFile()  must be of the type string, int given, called in  /var/www/html/nextcloud/lib/private/AppFramework/Http/Dispatcher.php on  line 169/var/www/html/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 169: OCA\Music\Controller\RadioApiController->exportAllToFile()/var/www/html/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 100: OC\AppFramework\Http\Dispatcher->executeController()/var/www/html/nextcloud/lib/private/AppFramework/App.php - line 152: OC\AppFramework\Http\Dispatcher->dispatch()/var/www/html/nextcloud/lib/private/Route/Router.php - line 309: OC\AppFramework\App::main()/var/www/html/nextcloud/lib/base.php - line 1008: OC\Route\Router->match()/var/www/html/nextcloud/index.php - line 37: OC::handleRequest()
-- | --

**Comment by MattRC7** [comment]
I also encountered this issue. I noticed when I entered the folder where I wanted to export the playlist to, it said there were no files in there, even though I double-checked and I have several .pls files in that folder. It still let me click "Choose" without selecting a file, and that's when the `Unexpected error` occurred.

Hope this helps:

![image](https://user-images.githubusercontent.com/6463042/107411256-51a43f00-6adc-11eb-8cba-9861e637f396.png)

**Comment by paulijar** [comment]
Okay, I could reproduce the error now on NC 20.0.7. The API of the Nextcloud file picker seems to have changed a bit, and this now breaks the exporting functionality. The problem affects also playlist exporting. I'm not sure, when has it broken, but I *think* I still successfully used export on some earlier hotfix of NC20.

@MattRC7 Not showing any files there is normal, we are using the filepicker dialog in folder selection mode. Hence, it shows only folders and no files. The text "No files in here" is quite misleading, though.

**Comment by paulijar** [comment]
Upon closer examination, it was revealed that the playlist and radio exporting is broken on Music v1.0.0 with all the supported Nextcloud versions except 13. There is an API difference between ownCloud and Nextcloud, which I hadn't taken into account, but the exporting had worked  also on NC regardless, simply by chance. On v1.0.0, we ran out of luck and the bug became apparent.

Fix for this is now implemented, and I'm planning to make a hotfix release soon.
